core/ledger/kvledger/upgrade_dbs.go

```diff
@@ -24,29 +32,39 @@ func UpgradeDBs(rootFSPath string) error {
 	defer fileLock.Unlock()
 
 	logger.Infof("Ledger data folder from config = [%s]", rootFSPath)
+	dbPath := LedgerProviderPath(rootFSPath)
+	db := leveldbhelper.CreateDB(&leveldbhelper.Conf{DBPath: dbPath})
+	db.Open()
+	defer db.Close()
 
-	// upgrades idStore (ledgerProvider) and drop databases
-	if err := UpgradeIDStoreFormat(rootFSPath); err != nil {
+	idStore := &idStore{db, dbPath}
+	format, err := idStore.getFormat()
```

Comment:
before checking for the format, shouldn't we check whether the `idStore` is empty and return early. It is a remote case though.

Reply:
Good point. This is like a user error. idStore cannot be empty unless something is wrong, e.g., ledger is corrupted or fileSystemPath in core.yaml points to a wrong path.